spec/grape/dsl/inside_route_spec.rb

```diff
@@ -111,11 +111,42 @@ def initialize
       expect(subject.status).to eq 204
     end
 
-    it 'defaults to 200 on DELETE with a body present' do
-      request = Grape::Request.new(Rack::MockRequest.env_for('/', method: 'DELETE'))
-      subject.body 'content here'
-      expect(subject).to receive(:request).and_return(request)
-      expect(subject.status).to eq 200
+    describe 'what is regarded as content on DELETE' do
+      it 'regards a string as content' do
+        request = Grape::Request.new(Rack::MockRequest.env_for('/', method: 'DELETE'))
+        subject.body 'content here'
+        expect(subject).to receive(:request).and_return(request)
+        expect(subject.status).to eq 200
+      end
+
+      it 'regards an empty string as content' do
+        request = Grape::Request.new(Rack::MockRequest.env_for('/', method: 'DELETE'))
+        subject.body ''
+        expect(subject).to receive(:request).and_return(request)
+        expect(subject.status).to eq 200
+      end
+
+      it 'regards an empty hash as content' do
+        request = Grape::Request.new(Rack::MockRequest.env_for('/', method: 'DELETE'))
+        subject.body({}) # Use brackets, because it's a block otherwise
+        expect(subject).to receive(:request).and_return(request)
+        expect(subject.status).to eq 200
+      end
+
+      it 'regards an empty array as content' do
+        request = Grape::Request.new(Rack::MockRequest.env_for('/', method: 'DELETE'))
+        subject.body []
+        expect(subject).to receive(:request).and_return(request)
+        expect(subject.status).to eq 200
+      end
+
+      # Still failing for some weird kind of reason.
+      it 'regards a nil as no content' do
+        request = Grape::Request.new(Rack::MockRequest.env_for('/', method: 'DELETE'))
+        subject.body nil
+        expect(subject).to receive(:request).and_return(request)
```

Comment:
This test doesn't work because there is wrong code in using `expect ~ receive`.
Depending on the implementation you modified, in this case [the `status` method is called internally](https://github.com/basjanssen/grape/blob/44cd8fb8d3fd84552c9f1c9748b683ce2a5da7e2/lib/grape/dsl/inside_route.rb#L232).
As a result, [the `@status` variable is assigned](https://github.com/basjanssen/grape/blob/44cd8fb8d3fd84552c9f1c9748b683ce2a5da7e2/lib/grape/dsl/inside_route.rb#L174) and [the `request` member isn't called and it ends](https://github.com/basjanssen/grape/blob/44cd8fb8d3fd84552c9f1c9748b683ce2a5da7e2/lib/grape/dsl/inside_route.rb#L176).

You don't need test this expectation. Please remove this line.